src/ape/api/compiler.py

```diff
@@ -75,12 +75,36 @@ def compile(
             List[:class:`~ape.type.contract.ContractType`]
         """
 
+    @raises_not_implemented
+    def compile_code(  # type: ignore[empty-body]
+        self,
+        code: str,
+        base_path: Optional[Path] = None,
+        contract_type_overrides: Optional[Dict] = None,
```

Comment:
what do you think about this?
```suggestion
        **contract_type_overrides: dict,
```

Reply:
it works for now

Reply:
the type is wrong though

Reply:
actually - the reason i didnt do this (or changed it from this) is because this would require the user to always pass `base_path` when only doing some overrides

Reply:
oh wait nvm, im mistaken

Reply:
I made this change, but it is kinda be funny looking in practice, because it looks like we have a mix of snake case and camel case kwargs:

```python
contract_type = self.compile_code(
    path.read_text(),
    base_path=base_path,
    sourceId=source_id,
)
```